### PR TITLE
[IMG-178] Linux vm support policy

### PR DESCRIPTION
**Pull request by JalexChen**
# Description
A brief description of the changes.
Adds Linux Image Support Policy to docs

# Reasons
[IMG-178](https://circleci.atlassian.net/jira/software/c/projects/IMG/boards/683?selectedIssue=IMG-178)

- [x] Break up walls of text by adding paragraph breaks.
- [x] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [x] Keep the title between 20 and 70 characters.
- [x] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [x] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [N/A] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [x] Include relevant backlinks to other CircleCI docs/pages.

[IMG-178]: https://circleci.atlassian.net/browse/IMG-178?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JalexChen** [comment]
@sebastian-lerner @brentmmarks @ryanbourdais - could i please get some eyes on this?

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 1]
```diff
@@ -0,0 +1,97 @@
+---
+contentTags:
+  platform:
+  - Cloud
+  - Server v4.x
```
Comment (brentmmarks) [inline jekyll/_cci2/linux-vm-support-policy.adoc:5]: Same server issue/question here

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+:page-layout: classic-docs
+:page-liquid:
+:page-description: CircleCI Linux VM images release, update, and deprecation policy
+:icons: font
+:experimental:
+
+[#overview]
+== Overview
+
+This document outlines the CircleCI Linux VM (Machine) Image release, update and deprecation policy. This policy applies to all CircleCI Linux VM (Machine) Images built for the Machine executor. Note that there is a separate [CUDA image support policy](https://circleci.com/docs/linux-cuda-images-support-policy/).
```
Comment (brentmmarks) [inline jekyll/_cci2/linux-vm-support-policy.adoc:18]: This does not align with what we got approved

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+The CircleCI Linux VM images are all based on Ubuntu, we support both ARM64 and AMD64 architectures.
+
+We aim to maintain, support, and release the two most recent LTS versions of Ubuntu.
+
+[#tagging]
+== Tagging
+
+Tags we support for this image (tag is what is specified in .circleci/config.yml when requesting an image):
+
+- `default`: This image is the current stable image. Jobs will default to this if no image or tag is specified.
```
Comment (brentmmarks) [inline jekyll/_cci2/linux-vm-support-policy.adoc:32]: Differs from approved

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+[#critical-cve-patches]
+== Critical CVE patches
+
+When critical CVEs are disclosed that affect the versions of the operating system or software stack in our remote Docker images, we will investigate the impact that this has on our images being used within the CircleCI execution environment.  If customers are impacted by these CVEs we will push a patch fix to the released image(s), this image will supersede the original image.
+
+[#bug-reports-issues-and-prs]
+== Bug reports, issues, and PRs
+
+You can file a link:https://support.circleci.com/hc/en-us/requests/new[support ticket] with CircleCI Support for any issues or bugs found with the remote Docker images. Our support team will be able to escalate issues internally to the correct engineering team and provide updates on the issue.
```
Comment (brentmmarks) [inline jekyll/_cci2/linux-vm-support-policy.adoc:52]: This is not remote docker image

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+When Ubuntu 24.04 LTS is released:
+
+[.table.table-striped]
+[cols=2*, options="header", stripes=even]
+|===
+| Version
+| Support
+
+| Ubuntu 20.04 LTS
+| Deprecated and removed
```
Comment (brentmmarks) [inline jekyll/_cci2/linux-vm-support-policy.adoc:81]: Deprecated and removed entirely

File: jekyll/_cci2/linux-vm-support-policy.adoc (hunk 2 above)
Comment (JalexChen) [reply jekyll/_cci2/linux-vm-support-policy.adoc:18]: i think it is important to include references and backlinks to a very similar image as this eliminates confusion for all parties and is part of the PR template checklist. I can bring this up with support or maybe Sebastian can chime in here?

File: jekyll/_cci2/linux-vm-support-policy.adoc (hunk 3 above)
Comment (JalexChen) [reply jekyll/_cci2/linux-vm-support-policy.adoc:32]: will do

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+[#bug-reports-issues-and-prs]
+== Bug reports, issues, and PRs
+
+You can file a link:https://support.circleci.com/hc/en-us/requests/new[support ticket] with CircleCI Support for any issues or bugs found with the remote Docker images. Our support team will be able to escalate issues internally to the correct engineering team and provide updates on the issue.
+
+[#image-lifespan-eol]
+== Image lifespan / EOL
+
+mage Lifespan will generally follow the support status of Ubuntu by Canonical. Once a new Ubuntu LTS is released, we will begin the deprecation process of the oldest version that we support and schedule it to be removed. This allows us to maintain our aim of two LTS Ubuntu versions effectively.
```
[the remark this answers is not in the document]
Comment (JalexChen) [reply jekyll/_cci2/linux-vm-support-policy.adoc:57]: will do

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+| Support
+
+| Ubuntu 20.04 LTS
+| Deprecated and removed
+
+| Ubuntu 22.04 LTS
+| Retain only Q4 (October) releases from past two years
+
+| Ubuntu 24.04 LTS
+| Build quarterly images for current year and retain Q4 (October) release
```
[the remark this answers is not in the document]
Comment (JalexChen) [reply jekyll/_cci2/linux-vm-support-policy.adoc:87]: will do

File: jekyll/_cci2/linux-vm-support-policy.adoc (hunk 2 above)
Comment (brentmmarks) [reply jekyll/_cci2/linux-vm-support-policy.adoc:18]: I am ok if @sebastian-lerner is ok for this one

Comment (sebastian-lerner) [reply jekyll/_cci2/linux-vm-support-policy.adoc:16]: sorry, what's the confusion? what did get approved and what's the delta?

Comment (JalexChen) [reply jekyll/_cci2/linux-vm-support-policy.adoc:16]: @sebastian-lerner: the only delta from what was approved is the the addition of this line to differentiate this from the Linux CUDA image

`Note that there is a separate [CUDA image support policy](https://circleci.com/docs/linux-cuda-images-support-policy/).`

Comment (sebastian-lerner) [reply jekyll/_cci2/linux-vm-support-policy.adoc:16]: okay seems fine to me

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+:page-layout: classic-docs
+:page-liquid:
+:page-description: CircleCI Linux VM images release, update, and deprecation policy
+:icons: font
+:experimental:
+
+[#overview]
+== Overview
+
+This document outlines the CircleCI Linux VM (Machine) Image release, update and deprecation policy. This policy applies to all CircleCI Linux VM (Machine) Images built for the Machine executor. Note that there is a separate [CUDA image support policy](https://circleci.com/docs/linux-cuda-images-support-policy/).
```
Comment (brentmmarks) [inline jekyll/_cci2/linux-vm-support-policy.adoc:16]: Please fix link format

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+Image Lifespan will generally follow the support status of Ubuntu by Canonical. Once a new Ubuntu LTS is released, we will begin the deprecation process of the oldest version that we support and schedule it to be removed. This allows us to maintain our aim of two LTS Ubuntu versions effectively.
+
+Current Deprecation (Jan 2024):
+
+[.table.table-striped]
+[cols=2*, options="header", stripes=even]
+|===
+| Version
+| Support
+
```
Comment (brentmmarks) [inline jekyll/_cci2/linux-vm-support-policy.adoc:64]: Missing Ubuntu 22.04 LTS

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+| Ubuntu 20.04 LTS
+| Deprecated and removed entirely
+
+| Ubuntu 22.04 LTS
+| Retain only Q4 (October) releases from past two years
+
+| Ubuntu 24.04 LTS
+| Quarterly images biuld and only Q4 (October) image retained from each year
```
Comment (brentmmarks) [inline jekyll/_cci2/linux-vm-support-policy.adoc:85]: spelling mistake in biuld -> build

**Review by brentmmarks — APPROVED** [review]

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+:page-layout: classic-docs
+:page-liquid:
+:page-description: CircleCI Linux VM images release, update, and deprecation policy
+:icons: font
+:experimental:
+
+[#overview]
+== Overview
+
+This document outlines the CircleCI Linux VM (Machine) Image release, update and deprecation policy. This policy applies to all CircleCI Linux VM (Machine) Images built for the Machine executor. Note that there is a separate link:https://circleci.com/docs/linux-cuda-images-support-policy/[CUDA image support policy].
```
Comment (rosieyohannan) [inline jekyll/_cci2/linux-vm-support-policy.adoc:16]:
```suggestion
This document outlines the CircleCI Linux VM (machine) image release, update and deprecation policy. This policy applies to all CircleCI Linux VM (machine) images built for the machine executor. Note that there is a separate xref:linux-cuda-images-support-policy#[CUDA image support policy].
```

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+== Release policy
+
+The CircleCI Linux VM images are all based on Ubuntu, we support both ARM64 and AMD64 architectures.
+
+We aim to maintain, support, and release the two most recent LTS versions of Ubuntu.
+
+[#tagging]
+== Tagging
+
+Tags we support for this image (tag is what is specified in .circleci/config.yml when requesting an image):
```
Comment (rosieyohannan) [inline jekyll/_cci2/linux-vm-support-policy.adoc:28]:
```suggestion
Tags we support for this image (tag is what is specified in `.circleci/config.yml` when requesting an image):
```

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+
+- `default` version is what you will receive if do not specify an image.
+
+- `current` version of image will receive updates approximately every 3 months.
```
Comment (rosieyohannan) [inline jekyll/_cci2/linux-vm-support-policy.adoc:32]:
```suggestion
- `current` version of the image will receive updates approximately every three months.
```

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+
+- `previous` version of image will receive the previous version of current.
```
Comment (rosieyohannan) [inline jekyll/_cci2/linux-vm-support-policy.adoc:34]:
```suggestion
- `previous` version of the image will receive the previous version of `current`.
```

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+
+- `edge` tags are reserved for previews of new releases, which will initially point to this tag. The edge tags may include incremental updates to the current quarterly (every 3 months) image release, which may change without notice, and is not recommended to be used for production CI workloads. currentwill be updated with these changes after a period of stability.
```
Comment (rosieyohannan) [inline jekyll/_cci2/linux-vm-support-policy.adoc:36]:
```suggestion
- `edge` tags are reserved for previews of new releases, which will initially point to this tag. The edge tags may include incremental updates to the current quarterly (every 3 months) image release, which may change without notice, and is not recommended to be used for production CI workloads. `current` will be updated with these changes after a period of stability.
```

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+
+- `Date based tagging`: The Ubuntu LTS based images have the image slug format of: `ubuntu-<LTS version>:<year>.<month>.<patch>`, such as `ubuntu-2204:2023.10.1`. We recommend using the `default` version and not pinning to a date version.
```
Comment (rosieyohannan) [inline jekyll/_cci2/linux-vm-support-policy.adoc:38]:
```suggestion
- Date based tagging: The Ubuntu LTS based images have the image slug format of: `ubuntu-<LTS version>:<year>.<month>.<patch>`, such as `ubuntu-2204:2023.10.1`. We recommend using the `default` version and not pinning to a date version.
```

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+
+We always aim to release a version of this image every 3 months.
```
Comment (rosieyohannan) [inline jekyll/_cci2/linux-vm-support-policy.adoc:40]:
```suggestion
We always aim to release a version of this image every three months.
```

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+
+[#critical-cve-patches]
+== Critical CVE patches
+
+When critical CVEs are disclosed that affect the versions of the operating system or software stack in Linux VM images, we will investigate the impact that this has on our images being used within the CircleCI execution environment.  If customers are impacted by these CVEs we will push a patch fix to the released image(s), this image will supersede the original image.
```
Comment (rosieyohannan) [inline jekyll/_cci2/linux-vm-support-policy.adoc:45]:
```suggestion
When critical CVEs are disclosed that affect the versions of the operating system or software stack in Linux VM images, we will investigate the impact that this has on our images being used within the CircleCI execution environment.  If customers are impacted by these CVEs we will push a patch fix to the released image(s). This image will supersede the original image.
```

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+Current Deprecation (Jan 2024):
+
+[.table.table-striped]
+[cols=2*, options="header", stripes=even]
+|===
+| Version
+| Support
+
+| Ubuntu 20.04 LTS
+| Build quarterly images for current year and retain Q4 (October) release from past two years.
```
Comment (rosieyohannan) [inline jekyll/_cci2/linux-vm-support-policy.adoc:66]:
```suggestion
| Build quarterly images for current year and retain Q4 (October) release from the past two years.
```

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 20]
```diff
@@ -0,0 +1,98 @@
...
+
+| Ubuntu 22.04 LTS
+| Quarterly images built and only Q4 image retained each year
+|===
+
+
+When Ubuntu 24.04 LTS is released:
```
Comment (rosieyohannan) [inline jekyll/_cci2/linux-vm-support-policy.adoc:73]:
```suggestion
Example: When Ubuntu 24.04 LTS is released
```

File: jekyll/_cci2/linux-vm-support-policy.adoc [hunk 21]
```diff
@@ -0,0 +1,98 @@
...
+| Ubuntu 22.04 LTS
+| Retain only Q4 (October) releases from past two years
+
+| Ubuntu 24.04 LTS
+| Quarterly images built and only Q4 (October) image retained from each year
+|===
+
+When an image is selected for deprecation and removal, we will create an announcement on our Discuss forum, along with reaching out via email to developers who have requested one of the deprecated images in their recent jobs.
+
+We will also plan brownouts to ensure users are aware of the approaching removal of deprecated images. Generally, we will aim to start an EOL process within 3 months of a new version release
```
Comment (rosieyohannan) [inline jekyll/_cci2/linux-vm-support-policy.adoc:93]:
```suggestion
We will also plan brownouts to ensure users are aware of the approaching removal of deprecated images. Generally, we will aim to start an EOL process within 3 months of a new version release.
```

**Pull request merged**